File: ScheduleFaculty.Core/Entities/Course.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace ScheduleFaculty.Core.Entities;

public class Course
{
    public Guid Id { get; set; }
    
    [ForeignKey(("StudyProgram"))]
    public Guid StudyProgramYearId { get; set; }
    public StudyProgram StudyProgram { get; set; }
    
    [ForeignKey(("ProfessorUser"))]
    public Guid ProfessorUserId { get; set; }
    public ApplicationUser ProfessorUser { get; set; }
    
    public string Name { get; set; }
    
    public string Abbreviation { get; set; }
    
    public int Semester { get; set; }
    
    public bool IsOptional { get; set; }
    
    
}

File: Frontend/src/app/models/course.ts
import {IStudyProgram} from "./study-program";
import {IUser} from "./login";
import {IHourType} from "./hour-type";

export interface ICourse {
  id?: string | null,
  studyProgramYearId: string | null,
  professorUserId: string | null,
  name: string | null,
  abbreviation: string | null,
  semester: number | null,
  isOptional: boolean | null
}

export interface ICourseResponse {
  id: string,
  studyProgram: IStudyProgram,
  professorUser: IUser,
  hourTypes: IHourType[],
  name: string | null,
  abbreviation: string | null,
  semester: number | null,
  isOptional: boolean | null
}



